pkg/util/workqueue/timed_queue.go

```diff
@@ -39,24 +39,28 @@ func NewTimedWorkQueue(clock clock.Clock) *TimedWorkQueue {
 // Add adds the obj along with the current timestamp to the queue.
 func (q TimedWorkQueue) Add(obj interface{}) {
 	start := q.clock.Now()
```

Comment:
As long as Get() and Done() are type safe, there should be no memory leak.

Using `obj interface{}` here is no worse than the regular work queue. Caller of Get() should check the type.

I'm trying to make caller's life easier here, they don't need to add timestamp.